DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/ManageCourseController.cs

```diff
@@ -41,5 +43,87 @@ public IActionResult Index(int customisationId)
 
             return View(model);
         }
+
+        [HttpGet]
+        [Route("LearningPathwayDefaults")]
+        public IActionResult EditLearningPathwayDefaults(int customisationId)
+        {
+            var centreId = User.GetCentreId();
+            var categoryId = User.GetAdminCategoryId()!;
+
+            var courseDetails = courseDataService.GetCourseDetailsForAdminCategoryId(
+                customisationId,
+                centreId,
+                categoryId.Value
+            );
+
+            if (courseDetails == null)
+            {
+                return NotFound();
+            }
+
+            var model = new EditLearningPathwayDefaultsViewModel(courseDetails);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("LearningPathwayDefaults")]
+        public IActionResult SaveLearningPathwayDefaults(
+            int customisationId,
+            EditLearningPathwayDefaultsViewModel model
+        )
+        {
+            if (customisationId != model.CustomisationId)
+            {
+                return new StatusCodeResult(500);
+            }
+
+            ValidateNumberInput(model.CompleteWithinMonths);
+            ValidateNumberInput(model.ValidityMonths);
+
+            if (!ModelState.IsValid)
+            {
+                return View("EditLearningPathwayDefaults", model);
+            }
+
+            if (model.AutoRefresh)
+            {
+                // Redirect to "Edit auto-refresh options" page
+                // To be configured in HEEDLS-442
+            }
+
+            var completeWithinMonthsInt =
+                model.CompleteWithinMonths == null ? 0 : int.Parse(model.CompleteWithinMonths);
+            var validityMonthsInt =
+                model.ValidityMonths == null ? 0 : int.Parse(model.ValidityMonths);
+
+            courseService.UpdateLearningPathwayDefaultsForCourse(
+                model.CustomisationId,
+                completeWithinMonthsInt,
+                validityMonthsInt,
+                model.Mandatory,
+                model.AutoRefresh
+            );
+
+            return RedirectToAction("Index", new { customisationId = model.CustomisationId });
+        }
+
+        private void ValidateNumberInput(string? numberInput)
+        {
+            if (numberInput == null)
+            {
+                return;
+            }
+
+            if (!int.TryParse(numberInput, out _))
+            {
+                ModelState.AddModelError(nameof(numberInput), "Value must only contain numbers");

```

Comment:
I think the error message here could be replaced with the message for the other error. I think the inclusion of "0" and "48" helps indicate the correct formatting, and the wording is general enough to apply.

Reply:
I agree, changed :)